connectors/filtering/basic_rule.py

```diff
@@ -114,13 +117,22 @@ def __eq__(self, other):
 
 
 class BasicRuleEngine:
+    """Class executing a list of basic rules in order against a document."""
+
     def __init__(self, rules):
         self.rules = rules
         self.rules_match_stats = {
             BasicRule.DEFAULT_RULE_ID: RuleMatchStats(Policy.INCLUDE, 0)
         }
 
     def should_ingest(self, document):
+        """Method to check, whether a document should be ingested or not.
+
+        On default the document will be ingested, if it doesn't match any rule.
```

Comment:
```suggestion
        By default the document will be ingested, if it doesn't match any rule.
```

What about "not by default" case?